Python/Product/IronPython/Interpreter/RemoteInterpreter.cs

```diff
@@ -1055,9 +1055,10 @@ private class CannotUnwrapHandleException : Exception {
                 if (genType != null) {
                     Type[] genTypes = new Type[types.Length];
                     for (int i = 0; i < types.Length; i++) {
-                        genTypes[i] = (Type)Unwrap(types[i]);
+                        var o = Unwrap(types[i]);
+                        genTypes[i] = o as Type ?? (PythonType)o;

```

Comment:
It has an implicit cast to Type that isn't handled by the `as` cast (C# won't do object->PythonType->Type in one step).